Feature/homework1

SharkoYaroslavQA19 opened the pull request.

SharkoYaroslavQA19 committed:
Salesforce homework part 1

SharkoYaroslavQA19 committed:
Salesforce homework part 1

SharkoYaroslavQA19 committed:
Salesforce homework part 2
added logging
added PageFactory
usage lombok

voropa commented on src/test/java/PageFactory/FactoryAccount.java:
```diff
@@ -0,0 +1,48 @@
+public class FactoryAccount {
+    Faker faker = new Faker();
+
+    public Account AccountWithAllData() {
```
Сделайте методы в этих классах  статическими, чтобы не нужно  было создавать экземпляры этих классов в тестах

voropa replied:
Названия методов с маленькой буквы

voropa commented on src/test/java/tests/CreateAccountsTest.java:
```diff
@@ -0,0 +1,40 @@
+        Assert.assertEquals(HomePage.getMessageText(), Message.expectedAccountMessageText(newAccount.getAccountName()));
+        AccountDetailsPage.getAccountInfo();
+        Assert.assertEquals(AccountDetailsPage.getAccountInfo(), newAccount);
+        HomePage.LogOut();
```
Как минимум нужно эти действия вынести в afterMethod, так как они у вас повторяются во  всех  тестах, а как максимум нужно убрать thread.sleep() и заменить его на explicitlyWait

voropa commented on src/test/java/tests/TestListener.java:
```diff
@@ -0,0 +1,31 @@
+    @Override
+    public void onTestFailure(ITestResult result) {
+        ITestListener.super.onTestFailure(result);
+        System.out.println("Test " + result.getName() + " failed");
```
Хорошо бы заменить на log.debug, раз вы используете логгер

voropa commented on src/test/java/pages/HomePage.java:
```diff
@@ -0,0 +1,67 @@
+
+        driver.findElement(LOGOUT_LINK).click();
+    }
+    public void LogOut() {
```
Названия методов с маленькой буквы

SharkoYaroslavQA19 committed:
Salesforce homework part 3
1. FactoryPage converted to static methods
2. removed from thread.sleep() tests, logout moved to afterMethod
3. added logdebug to Testlistener

SharkoYaroslavQA19 committed:
Salesforce homework part 3
1. FactoryPage converted to static methods
2. removed from thread.sleep() tests, logout moved to afterMethod
3. added logdebug to Testlistener
